### PR TITLE
Issue #424.

### PR DESCRIPTION
**Pull request by temuri416**
Allow custom DB handler via Behaviour's constructor.

### CODE REVIEW

File: Library/Phalcon/Mvc/Model/Behavior/NestedSet.php [hunk 1]
```diff
@@ -6,9 +6,14 @@
 use Phalcon\Mvc\Model\BehaviorInterface;
 use Phalcon\Mvc\Model;
 use Phalcon\Mvc\ModelInterface;
+use Phalcon\Db\Adapter as dbAdapter;
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: Can you please rename alias to ` DbAdapter`?

File: Library/Phalcon/Mvc/Model/Behavior/NestedSet.php [hunk 2]
```diff
@@ -6,9 +6,14 @@
...
 
 class NestedSet extends Behavior implements BehaviorInterface
 {
+    /**
+     * @var dbAdapter
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: DbAdapter|null

File: Library/Phalcon/Mvc/Model/Behavior/NestedSet.php [hunk 3]
```diff
@@ -67,6 +76,14 @@ public function missingMethod(ModelInterface $model, $method, $arguments = null)
             return null;
         }
 
+        if (!$this->db) {
+            if ($model->getDi()->has('db')) {
+                $this->db = $model->getDi()->get('db');
+            } else {
+                throw new \Exception('Undefined database handler.');
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: add please `@throws` annotation

Comment (temuri416) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: Where do you want it - in a docblock of the constructor?

Comment (temuri416) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: added docblock to the constructor.

**Comment by sergeyklay** [comment]
Now when everything is fine, could you rebase and squash it into a single commit to keep the history clean? Let me know if you need instructions.

File: Library/Phalcon/Mvc/Model/Behavior/NestedSet.php (hunk 3 above)
Comment (sergeyklay) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: How about something like this? :grinning: 

```php
$this->db = $this->db ?: $model->getDi()->has('db') && $model->getDi()->get('db');

if (!$this->db) {
    throw new \Exception('Undefined database handler.');
}
```

Comment (temuri416) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: Sure. I avoid nested ternaries because execution path is not clear when using remote debugging.

Comment (sergeyklay) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: Okay. Now just squash please all commits into a single commit to keep the history clean.

Comment (temuri416) [inline Library/Phalcon/Mvc/Model/Behavior/NestedSet.php]: Done. Let me know if I squashed smth wrong :)

**Comment by sergeyklay** [comment]
Thanks!

**Pull request merged**